### PR TITLE
[logging] Add logging functionality

### PR DESCRIPTION
**Pull request by falkecarlsen**
Logging now works by using `logging.{debug, info, warning, error, critical}(msg)` and writes nicely formatted to stdout and file.

Existing prints and important places of debug, info, and warning has been converted to logging.

Note that log-levels are symmetric between stdout and file which is unintended, as logfile should always be verbose. This allows a user to look up a previous run if required. I was not able to solve it easily and have left a stack-link if we decide to pursue that feature further.

### CODE REVIEW

**Comment by falkecarlsen** [comment]
Generally, follow this principle when choosing log level:
- Debug: Data being used for this procedure.
- Info: What is being done.
- Warning: Non-breaking faults.
- Error: Breaking faults.
- Critical: [You've got bigger problems](https://www.explainxkcd.com/wiki/index.php/2200:_Unreachable_State).

File: randomActivityGen.py [hunk 1]
```diff
@@ -225,36 +236,78 @@ def verify_stats(stats: ET.ElementTree):
...
         setup_schools(net, stats, int(args["--schools.count"]))
 
     # Write statistics back
+    logging.info(f"Writing statistics file to {args['--output-file']}")
     stats.write(args["--output-file"])
 
     if args["--display"]:
-        display_network(net, stats, 500, 500)
+        x_size, y_size = 500, 500
+        logging.info(f"Displaying network as image sized: {x_size} x {y_size}")
```
Comment (NicEastvillage) [inline randomActivityGen.py]: The produced image is not guaranteed to be `x_size` x `y_size`. The arguments are *max* width and height as the city might not be quadratic.

Comment (falkecarlsen) [reply randomActivityGen.py]: Yep, I forgot that. Fixed in 39bfdb7

**Review by NicEastvillage — COMMENTED** [review]
I think the program should say "Done" when it succesfully finish ;)

**Comment by falkecarlsen** [comment]
> I think the program should say "Done" when it succesfully finish ;)

I disagree, it ends with outputting "Writing stats to..." which is quite clear.
image

*However, the display option could be moved above the writing of stats, to ensure that is always the last written line.*

**Comment by NicEastvillage** [comment]
Meh. The program can fail after printing the "Writing stats to...", but whatever. I don't mind display being after writing output.

**Review by NicEastvillage — APPROVED** [review]

**Comment by falkecarlsen** [comment]
> Meh. The program can fail after printing the "Writing stats to...", but whatever. I don't mind display being after writing output.

That's what error codes are for :)

**Review by MathiasMehl — APPROVED** [review]

**Pull request merged**